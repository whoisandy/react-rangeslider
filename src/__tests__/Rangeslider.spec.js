import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import Slider from '../Rangeslider'

describe('Rangeslider specs', () => {
  it('should render properly', () => {
    const slider = shallow(<Slider />)
    expect(slider.hasClass('rangeslider')).toBeTruthy()
    expect(slider.children().length).toEqual(3)
    expect(slider.find('.rangeslider__fill').length).toEqual(1)
    expect(slider.find('.rangeslider__handle').length).toEqual(1)
  })

  it('should have default props', () => {
    const slider = mount(<Slider />)
    expect(slider.prop('min')).toEqual(0)
    expect(slider.prop('max')).toEqual(100)
    expect(slider.prop('step')).toEqual(1)
    expect(slider.prop('value')).toEqual(0)
    expect(slider.prop('orientation')).toEqual('horizontal')
    expect(slider.prop('reverse')).toEqual(false)
    expect(slider.prop('handleLabel')).toEqual('')
    expect(slider.prop('labels')).toEqual({})
    expect(slider.prop('disabled')).toEqual(false)
  })

  it('should render basic slider with defaults', () => {
    const tree = renderer.create(<Slider />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render slider when props passed in', () => {
    const tree = renderer
      .create(<Slider min={10} max={50} value={20} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
