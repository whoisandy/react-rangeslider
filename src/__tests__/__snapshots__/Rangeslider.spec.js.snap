// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rangeslider specs should render basic slider with defaults 1`] = `
<div
  aria-orientation="horizontal"
  aria-valuemax={100}
  aria-valuemin={0}
  aria-valuenow={0}
  className="rangeslider rangeslider-horizontal"
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="rangeslider__fill"
    style={
      Object {
        "width": "0px",
      }
    }
  />
  <div
    className="rangeslider__handle"
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    style={
      Object {
        "left": "0px",
      }
    }
    tabIndex={0}
  >
    <div
      className="rangeslider__handle-label"
    >
      
    </div>
  </div>
  <ul
    className="rangeslider__labels"
  />
</div>
`;

exports[`Rangeslider specs should render slider when props passed in 1`] = `
<div
  aria-orientation="horizontal"
  aria-valuemax={50}
  aria-valuemin={10}
  aria-valuenow={20}
  className="rangeslider rangeslider-horizontal"
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="rangeslider__fill"
    style={
      Object {
        "width": "0px",
      }
    }
  />
  <div
    className="rangeslider__handle"
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    style={
      Object {
        "left": "0px",
      }
    }
    tabIndex={0}
  >
    <div
      className="rangeslider__handle-label"
    >
      
    </div>
  </div>
  <ul
    className="rangeslider__labels"
  />
</div>
`;
